tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/access/MemberAccessTest.java

```diff
@@ -370,6 +370,23 @@ public void testInvalidMemberAccessOnStrings2() {
         BRunUtil.invoke(result, "testInvalidMemberAccessOnStrings2");
     }
 
+    @Test
+    public void testMemberAccessWithBinaryExprAsIndex() {
+        BRunUtil.invoke(result, "testMemberAccessWithBinaryExprAsIndex");
+    }
+
+    @Test
+    public void testMemberAccessWithGroupExprAsIndex() {
+        BRunUtil.invoke(result, "testMemberAccessWithGroupExprAsIndex");
+    }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+            expectedExceptionsMessageRegExp = ".*\\{ballerina/lang.array\\}IndexOutOfRange \\{\"message\":\"array " +
+                    "index out of range: index: 3, size: 3\"\\}.*")
+    public void testMemberAccessOutOfRangeWithBinaryExpr() {
```

Comment:
We can use `trap` with the access and assert this also in b7a itself.

Reply:
Will update